CMakeLists.txt

```diff
@@ -63,12 +58,7 @@ if (CMAKE_BUILD_TYPE STREQUAL "")
   add_definitions(-O3)
 endif()
 
-if($ENV{ROS_DISTRO} STREQUAL "humble")  # the ros2 humble requires c++17
```

Comment:
In addition, why is the restriction of ros version removed here?

Reply:
Im not sure, but as this software is C++17 compatible, @Timple set it c++17 as default.

For new ROS2 distros you would need to add it here (iron, jazzy and rolling) if new distros are released.

Also, as you are using cmake version > 3.1. Is better define these variables:
set (CMAKE_CXX_STANDARD 17)
https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html

Reply:
> In addition, why is the restriction of ros version removed here?

It's not removed, it is still implicitly there. But the statement overlooked `iron`, `jazzy` and `rolling`. Currently `humble` is the oldest supported distro at the moment: [ros_clock](https://dlu.github.io/ros_clock/index.html). So all can use `c++17`